### PR TITLE
DROOLS-5775: Test Scenario does not support nested Enum type attributes

### PR DESCRIPTION
**Pull request by yesamer**
Before this change, the editor didn't support a "nested" FactName (eg. `Customer.Category`).
This PR enables nested enum only in RULE Test Scenario. But the same logic will be used to handle imported decisions or input in DMN Test Scenario (eg `cc.DecisionName`) in a future PR. 
No impacts on Kogito side for this PR.

Nested Enum As Simple Type
Screenshot from 2020-11-13 17-28-44

Nested Enum as property
Screenshot from 2020-11-13 17-35-25

**JIRA**: [DROOLS-5775](https://issues.redhat.com/browse/DROOLS-5775)

**Referenced Pull Requests**:
* https://github.com/kiegroup/drools/pull/3201
* https://github.com/kiegroup/kie-wb-common/pull/3470
* https://github.com/kiegroup/drools-wb/pull/1440

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by jstastny-cz** [comment]
How does this fit together with #1434 ? If that revert is merged, we loose support even for simple enum properties.

**Comment by yesamer** [comment]
@jstastny-cz This PR extends `enum` supports, therefore I expect to have problems for these changes as well after merging #1434 . I'll try to get a look on that topic next week, if @Rikkola is still busy with other tasks.

**Comment by yesamer** [comment]
jenkins retest this please

**Comment by yesamer** [comment]
jenkins retest this please

**Comment by yesamer** [comment]
jenkins retest this please

**Review by Rikkola — APPROVED** [review]

**Review by Rikkola — CHANGES_REQUESTED** [review]
Code looks clean, but I think I made some mistake when compiling and was not able to use it like in the screenshot.

**Comment by yesamer** [comment]
@Rikkola DId you compile `kie-wb-common` PR?

**Review by karreiro — APPROVED** [review]
Great stuff, @yesamer! LGTM and works fine as well :-)

**Review by Rikkola — APPROVED** [review]
Rebuilt and still did not see it, but I suspect I'm just using it wrong. If 2 out of 3 had it working I'm sure the problem is behind the keyboard.

**Comment by yesamer** [comment]
jenkins retest this please

**Comment by jstastny-cz** [comment]
jenkins do fdb

**Comment by yesamer** [comment]
jenkins do fdb

**Comment by jstastny-cz** [comment]
jenkins do fdb

**Comment by jstastny-cz** [comment]
jenkins do FDB

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/AbstractDMODataManagementStrategy.java
Comment (jstastny-cz) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/AbstractDMODataManagementStrategy.java:161]: we don't need this assignment here, null would be a better choice IMO.

**Review by jstastny-cz — APPROVED** [review]
Just a minor comment.

**Comment by jstastny-cz** [comment]
I'll approve after I test with the FDB.

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/AbstractDMODataManagementStrategy.java
Comment (yesamer) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/AbstractDMODataManagementStrategy.java:161]: @jstastny-cz That is the default value for `factClassName` variable. If I put `null` here, I need to define an `else` statement in line 166, defining `factClassName` variable. I think semantically it's the same.

**Comment by yesamer** [comment]
jenkins do FDB

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/AbstractDMODataManagementStrategy.java
Comment (jstastny-cz) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/AbstractDMODataManagementStrategy.java:161]: Ah, I missed that, thanks.

**Comment by Ginxo** [comment]
jenkins retest this

**Comment by Ginxo** [comment]
jenkins do FDB

**Comment by jstastny-cz** [comment]
Even though FDB failed, the BC war is built there, sufficient to test.

**Review by jstastny-cz — APPROVED** [review]
Works as expected, thanks.

**Comment by yesamer** [comment]
jenkins do FDB

**Pull request merged**
